### ISSUE TITLE
[Feature Request] OnError extension

### ISSUE THREAD

**Issue by guffy1234** [danielgerlag/workflow-core#508]
hi

Probably it would be nice to have retry count and behavior after it reached zero.
I mean instead of 

IStepBuilder<TData, TStepBody> OnError(WorkflowErrorHandling behavior, TimeSpan? retryInterval = null);

have something like 

IStepBuilder<TData, TStepBody> OnError(WorkflowErrorHandling behavior, TimeSpan? retryInterval = null, int?  retryCount  = int.MaxValue, WorkflowErrorHandling? failBehavior = WorkflowErrorHandling.Terminate);

**Comment by danielgerlag** [comment]
Yes I agree, being able to define complex error strategies would be useful.

**Comment by wfcore-manager** [comment]
Closing as not planned — this has been open for several years without recent progress. Please open a new issue if this is still relevant on a current Workflow Core release.
